test/pummel/test-hash-seed.js

```diff
@@ -1,31 +1,25 @@
 'use strict';
 
 // Check that spawn child doesn't create duplicated entries
-require('../common');
-const Countdown = require('../common/countdown');
+const common = require('../common');
 const REPETITIONS = 2;
 const assert = require('assert');
 const fixtures = require('../common/fixtures');
-const { spawn } = require('child_process');
+const { promisify } = require('util');
+const { execFile } = require('child_process');
+const execFilePromise = promisify(execFile);
 const targetScript = fixtures.path('guess-hash-seed.js');
-const seeds = [];
 
-const requiredCallback = () => {
+const requiredCallback = common.mustCall((results) => {
+  const seeds = results.map((val) => val.stdout.trim());
   console.log(`Seeds: ${seeds}`);
   assert.strictEqual(new Set(seeds).size, seeds.length);
   assert.strictEqual(seeds.length, REPETITIONS);
-};
+});
 
-const countdown = new Countdown(REPETITIONS, requiredCallback);
+const generateSeed = () => execFilePromise(process.execPath, [targetScript]);
+const subprocesses = [...new Array(REPETITIONS)].map(generateSeed);
 
-for (let i = 0; i < REPETITIONS; ++i) {
-  let result = '';
-  const subprocess = spawn(process.execPath, [targetScript]);
-  subprocess.stdout.setEncoding('utf8');
-  subprocess.stdout.on('data', (data) => { result += data; });
-
-  subprocess.on('exit', () => {
-    seeds.push(result.trim());
-    countdown.dec();
-  });
-}
+Promise.all(subprocesses)
+  .then(requiredCallback)
+  .catch(assert.fail);
```

Comment:
The `catch` handler here is not necessary since the test will fail with an unhandled rejection by default already.

Reply:
Yes, but that's magic handled by `common` and I'd rather have this be a good example for people to copy/paste. I can remove it if that's an issue, though.

Reply:
Ok, consider it non-blocking.

Reply:
The error thrown generated by `assert.fail` is still magically caught by `common`.

Reply:
> The error thrown by `assert.fail` is still magically handled by `common`.

Doh!